pkg/finder/finder.go

```diff
@@ -464,3 +472,23 @@ func getContainerEnvironmentVariables(ci ContainerInfo) map[string]string {
 	// nothing else to try, give up and fail!
 	return nil
 }
+
+func getAllocatedCPUs(cgroupsPath string, CRIEndpointPath string) (string, error) {
```

Comment:
this function does two main things:
1. it finds the full cgroup path given 
a. the host sysfs mount path
b. the detected container runtime
2. reads the cpu allocation from the path learned from the step above

Now, turns out that everything is actually generic provided we translate the path we learn from the container info into a real path. Right now we learn about runtime guessing from the endpointpath. But nothing in the actual path except for the conventions guarantees that we can safely deduct the runtime from the path. In other words, nothing prevents someone to configure the kubelet to use something like `/run/cri/runtime.sock` independenty from crio or containerd.

TL;DR: I think we need an explicit flag, that's the only way we can really be sure.

Reply:
I understand, I wasn't a huge fan of using the socket names either, will add a flag for explicitly specifying the container runtime